srtp/srtp.c

```diff
@@ -279,6 +280,46 @@ static srtp_err_status_t srtp_stream_dealloc(
     return srtp_err_status_ok;
 }
 
+/* try to insert stream in list or deallocate it */
+static srtp_err_status_t srtp_insert_or_dealloc_stream(srtp_stream_list_t list,
+                                                       srtp_stream_t stream,
+                                                       srtp_stream_t template)
+{
+    srtp_err_status_t status = srtp_stream_list_insert(list, stream);
+    /* on failure, ownership wasn't transferred and we need to deallocate */
+    if (status)
+        srtp_stream_dealloc(stream, template);
```

Comment:
```suggestion
    if (status) {
        srtp_stream_dealloc(stream, template);
    }
```

Reply:
... and please fix similar instances below.

Reply:
we should really do a sweep of all the code and fix this every where. But agree that it is good add them as we go.